perf/run-benchmarks

```diff
@@ -453,6 +453,9 @@ def run_benchmarks(
         variants = GITLAB_VARIANTS
     elif small_only:
         corpuses = SMALL_CORPUSES
+    elif output_time_per_rule_json:
```

Comment:
Do we need this? If we use this flag, we can still use it with the other flags like `--std-only` and `--small-only` independent of whether it reports back average rule runtimes

Reply:
right, the reason why I added this was because I wanted to make sure that the `--output-times-per-json` flag is run with `--std-only`. The times per-repository aren't accurate without running `--std-only` (because it will only output the latest run for a specific repository, so if all the variants run then it will output the results for the last optimized run). The times per-rule also come out not completely accurate because they are added together then divided for each run 🤔 

Perhaps a better way to do this is to throw an error if `output-time-per-rule` isn't rule with `std-only`.

Reply:
Add a logging INFO line that reports that this flag will only run with std benchmarks